hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1427,7 +1487,7 @@ public void markContainerUnhealthy(Container container, ScanResult reason)
     } finally {
       container.writeUnlock();
     }
-    createContainerMerkleTreeFromMetadata(container);
+    updateContainerChecksumFromMetadataIfNeeded(container);
```

Comment:
https://github.com/apache/ozone/pull/8402#discussion_r2080078695 In #8402 we were planning to do an On-demand scanning here rather than generating the merkle tree from the metadata.

Reply:
I filed HDDS-13084 to do this in a follow-up. It will need testing that would expand this already large patch.